opds.py

```diff
@@ -607,6 +642,75 @@ def page(cls, _db, title, url, lane, annotator,
             return cached
         return content
 
+    @classmethod
+    def facet_link(cls, href, title, facet_group_name, is_active):
+        """Build a set of attributes for a facet link.
+
+        :param href: Destination of the link.
+        :param title: Human-readable description of the facet.
+        :param facet_group_name: The facet group to which the facet belongs,
+           e.g. "Sort By".
+        :param is_active: True if this is the client's currently
+           selected facet.
+
+        :return: A dictionary of attributes, suitable for passing as
+            keyword arguments into OPDSFeed.add_link_to_feed.
+        """
+        args = dict(href=href, title=title)
+        args['rel'] = cls.FACET_REL
+        args['{%s}facetGroup' % AtomFeed.OPDS_NS] = facet_group_name
+        if is_active:
+            args['{%s}activeFacet' % AtomFeed.OPDS_NS] = "true"
+        return args
+
+    @classmethod
+    def add_entrypoint_links(cls, feed, url_generator, entrypoints,
+                             selected_entrypoint, group_name='Start here'):
```

Comment:
If the client has tabs, I'm guessing this group name won't be displayed, except maybe for screen readers. But I'm not sure how this sounds if it shows up in a regular OPDS client that knows about facets but not entry points. I don't have another suggestion right now though.

Reply:
"Start here" is pretty bad but it's ultimately up to the application to decide what kind of thing an entry point is.

Reply:
I'd like generic OPDS clients to be able to do something reasonable without knowing about entry points, so maybe it does have to be up to the library. If they selected 'Ebooks' and 'Audio' as their entry points they could call it 'Formats'.

Reply:
Since the only EntryPoints we have right now are for formats, I'll make "Formats" the default here, rather than have a default nobody actually wants to use.